core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -1243,29 +1269,43 @@ object IO extends Serializable {
   )(release: A => IO[Nothing, _])(use: A => IO[E, B]): IO[E, B] =
     Ref[IO[Nothing, Any]](IO.unit).flatMap { m =>
       (for {
-        a <- acquire.flatMap(a => m.set(release(a)).const(a)).uninterruptibly
+        a <- acquire.flatMap(a => m.set(release(a)).const(a)).uninterruptible
         b <- use(a)
       } yield b).ensuring(IO.flatten(m.get))
     }
 
   /**
    * Apply the function fn to each element of the `Iterable[A]` and
-   * return the results in a new `List[B]`. For parallelism use `parTraverse`.
+   * return the results in a new `List[B]`. For parallelism use `foreachPar`.
    */
-  final def traverse[E, A, B](in: Iterable[A])(fn: A => IO[E, B]): IO[E, List[B]] =
+  final def foreach[E, A, B](in: Iterable[A])(fn: A => IO[E, B]): IO[E, List[B]] =
     in.foldRight[IO[E, List[B]]](IO.sync(Nil)) { (a, io) =>
-      fn(a).seqWith(io)((b, bs) => b :: bs)
+      fn(a).zipWith(io)((b, bs) => b :: bs)
     }
 
+  /**
+   * Alias for foreach
+   */
+  @deprecated("Use foreach", "scalaz-zio 0.6.0")
```

Comment:
deprecated warning is a pretty big punish for 'FP-sounding' names... Can't say I like this, just means I have to import cats syntax more often for FP names.

Reply:
`foreach` is the most descriptive name for what kind of `IO` this produces, so if you're writing "directly against an `IO`", use that
of course, if you're writing "against a typeclass" (is it *Applicative functor* in this case?), use `traverse` not  `foreach` (it's not even available, after all)

Reply:
Makes sense I guess

Reply:
I don't mind keeping them around forever, personally. Although I do think if you're using Cats or Scalaz, it's easy to call `io.sequence`, you get that from the syntax.

Reply:
yeah, the intention is to keeps this aliases around -- people discovering them easily is their whole point! :slightly_smiling_face: